src/compiler/checker.ts

```diff
@@ -6543,6 +6549,17 @@ namespace ts {
             }
         }
 
+        function signatureHasTypePredicate(signature: Signature): boolean {
+            return signature.resolvedTypePredicate !== undefined;
+        }
+
+        function getTypePredicateOfSignature(signature: Signature): TypePredicate | undefined {

```

Comment:
Shouldn't `undefined` be used for the most common case though? Usually no type predicate exists.